File: study/2022/05-W3/wg.shin.js
/*
Runtime: 75 ms, faster than 89.50% of JavaScript online submissions for Remove Duplicates from Sorted Array.
Memory Usage: 45.1 MB, less than 21.40% of JavaScript online submissions for Remove Duplicates from Sorted Array.
*/
const removeDuplicates = (nums) => {nums.splice(0, nums.length, ...(new Set(nums)))};
// 한줄로 풀 수 있을꺼 같아서 풀어봤습니다.
// 처음에는 map으로도 될 줄 알았는데 splice를 쓰지 않으면 원본 객체가 변경되지 않더군요.

File: study/2022/06-W4/wg.shin.py
class Solution:
    def searchInsert(self, nums: List[int], target: int):
        if len(nums) == 1:
            if target <= nums[0]:
                return 0
            return 1
        def sol(startIdx, endIdx):
            if endIdx - startIdx == 0:
                if target <= nums[startIdx]:
                    return startIdx
                return startIdx + 1
            if endIdx - startIdx == 1:
                if target <= nums[startIdx]:
                    return startIdx
                if target <= nums[startIdx + 1]:
                    return startIdx + 1
                return startIdx + 2
            midIdx = startIdx + (endIdx - startIdx)//2 
            if target > nums[midIdx]:
                return sol(midIdx + 1, endIdx)
            return sol(startIdx, midIdx)
        
        return sol(0, len(nums) - 1)
      
"""
Runtime: 54 ms, faster than 87.70% of Python3 online submissions for Search Insert Position.
Memory Usage: 15 MB, less than 6.21% of Python3 online submissions for Search Insert Position.
recursive func 의 경우 heapStack을 차지하기 때문에 메모리는 어쩔수 없는거같다..
"""
